\hypertarget{group__file__utils}{}\section{Functions to Read/\+Write several File Formats for R\+NA Sequences, Structures, and Alignments}
\label{group__file__utils}\index{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}}


Functions dealing with file formats for R\+NA sequences, structures, and alignments.  


Collaboration diagram for Functions to Read/\+Write several File Formats for R\+NA Sequences, Structures, and Alignments\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=327pt]{group__file__utils}
\end{center}
\end{figure}
\subsection*{Files}
\begin{DoxyCompactItemize}
\item 
file \hyperlink{commands_8h}{commands.\+h}
\begin{DoxyCompactList}\small\item\em Parse and apply different commands that alter the behavior of secondary structure prediction and evaluation. \end{DoxyCompactList}\item 
file \hyperlink{file__formats_8h}{file\+\_\+formats.\+h}
\begin{DoxyCompactList}\small\item\em Read and write different file formats for R\+NA sequences, structures. \end{DoxyCompactList}\item 
file \hyperlink{file__formats__msa_8h}{file\+\_\+formats\+\_\+msa.\+h}
\begin{DoxyCompactList}\small\item\em Functions dealing with file formats for Multiple Sequence Alignments (M\+SA) \end{DoxyCompactList}\item 
file \hyperlink{file__utils_8h}{file\+\_\+utils.\+h}
\begin{DoxyCompactList}\small\item\em Several utilities for file handling. \end{DoxyCompactList}\item 
file \hyperlink{ribo_8h}{ribo.\+h}
\begin{DoxyCompactList}\small\item\em Parse Ribo\+Sum Scoring Matrices for Covariance Scoring of Alignments. \end{DoxyCompactList}\end{DoxyCompactItemize}
\subsection*{Macros}
\begin{DoxyCompactItemize}
\item 
\#define \hyperlink{group__file__utils_ga79a23de2c7249f2cccd762e475c81859}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+C\+L\+U\+S\+T\+AL}~1U
\begin{DoxyCompactList}\small\item\em Option flag indicating ClustalW formatted files. \end{DoxyCompactList}\item 
\#define \hyperlink{group__file__utils_ga62be992445cd8ab2ad7a8fded944338b}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+S\+T\+O\+C\+K\+H\+O\+LM}~2U
\begin{DoxyCompactList}\small\item\em Option flag indicating Stockholm 1.\+0 formatted files. \end{DoxyCompactList}\item 
\#define \hyperlink{group__file__utils_gacf6274a2c825f34e4131404665b00604}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+F\+A\+S\+TA}~4U
\begin{DoxyCompactList}\small\item\em Option flag indicating F\+A\+S\+TA (Pearson) formatted files. \end{DoxyCompactList}\item 
\#define \hyperlink{group__file__utils_gab3186d774c01570d2a47d0308eec4927}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+M\+AF}~8U
\begin{DoxyCompactList}\small\item\em Option flag indicating M\+AF formatted files. \end{DoxyCompactList}\item 
\#define \hyperlink{group__file__utils_ga4acc255373831856a8417b68de8a94c0}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+D\+E\+F\+A\+U\+LT}
\begin{DoxyCompactList}\small\item\em Option flag indicating the set of default file formats. \end{DoxyCompactList}\item 
\#define \hyperlink{group__file__utils_ga229fb3778ecabea4782902b69fa48fd1}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+N\+O\+C\+H\+E\+CK}~4096U
\begin{DoxyCompactList}\small\item\em Option flag to disable validation of the alignment. \end{DoxyCompactList}\item 
\#define \hyperlink{group__file__utils_gabdc948f547e550125de3e7c65878400c}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+U\+N\+K\+N\+O\+WN}~8192U
\begin{DoxyCompactList}\small\item\em Return flag of \hyperlink{group__file__utils_gade4fa8136ebb2d0f7eb3f8b59a8658e3}{vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format()} to indicate unknown or malformatted alignment. \end{DoxyCompactList}\end{DoxyCompactItemize}
\subsection*{Functions}
\begin{DoxyCompactItemize}
\item 
int \hyperlink{group__file__utils_gad02d5d12bda54611c915a1019323b7be}{vrna\+\_\+file\+\_\+msa\+\_\+read} (const char $\ast$filename, char $\ast$$\ast$$\ast$names, char $\ast$$\ast$$\ast$aln, char $\ast$$\ast$id, char $\ast$$\ast$structure, unsigned int options)
\begin{DoxyCompactList}\small\item\em Read a multiple sequence alignment from file. \end{DoxyCompactList}\item 
int \hyperlink{group__file__utils_gad4203a438622b2df7bc2f16578d54799}{vrna\+\_\+file\+\_\+msa\+\_\+read\+\_\+record} (F\+I\+LE $\ast$fp, char $\ast$$\ast$$\ast$names, char $\ast$$\ast$$\ast$aln, char $\ast$$\ast$id, char $\ast$$\ast$structure, unsigned int options)
\begin{DoxyCompactList}\small\item\em Read a multiple sequence alignment from file handle. \end{DoxyCompactList}\item 
unsigned int \hyperlink{group__file__utils_gade4fa8136ebb2d0f7eb3f8b59a8658e3}{vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format} (const char $\ast$filename, unsigned int options)
\begin{DoxyCompactList}\small\item\em Detect the format of a multiple sequence alignment file. \end{DoxyCompactList}\item 
void \hyperlink{group__file__utils_ga4382a56d2fee9ed738364b99329edc7c}{vrna\+\_\+file\+\_\+copy} (F\+I\+LE $\ast$from, F\+I\+LE $\ast$to)\hypertarget{group__file__utils_ga4382a56d2fee9ed738364b99329edc7c}{}\label{group__file__utils_ga4382a56d2fee9ed738364b99329edc7c}

\begin{DoxyCompactList}\small\item\em Inefficient `cp\textquotesingle{}. \end{DoxyCompactList}\item 
char $\ast$ \hyperlink{group__file__utils_gad16c270b70a77c753088b29de8a802dc}{vrna\+\_\+read\+\_\+line} (F\+I\+LE $\ast$fp)
\begin{DoxyCompactList}\small\item\em Read a line of arbitrary length from a stream. \end{DoxyCompactList}\item 
int \hyperlink{group__file__utils_gacfa44a974272a4d42c5d391cc780a0ec}{vrna\+\_\+mkdir\+\_\+p} (const char $\ast$path)\hypertarget{group__file__utils_gacfa44a974272a4d42c5d391cc780a0ec}{}\label{group__file__utils_gacfa44a974272a4d42c5d391cc780a0ec}

\begin{DoxyCompactList}\small\item\em Recursivly create a directory tree. \end{DoxyCompactList}\item 
char $\ast$ \hyperlink{group__file__utils_ga8d0f8f84f1e2628172cf3c8648b2e507}{vrna\+\_\+basename} (const char $\ast$path)\hypertarget{group__file__utils_ga8d0f8f84f1e2628172cf3c8648b2e507}{}\label{group__file__utils_ga8d0f8f84f1e2628172cf3c8648b2e507}

\begin{DoxyCompactList}\small\item\em Extract the filename from a file path. \end{DoxyCompactList}\item 
char $\ast$ \hyperlink{group__file__utils_gae006d31b088d52d8aace86948def5fe8}{vrna\+\_\+dirname} (const char $\ast$path)\hypertarget{group__file__utils_gae006d31b088d52d8aace86948def5fe8}{}\label{group__file__utils_gae006d31b088d52d8aace86948def5fe8}

\begin{DoxyCompactList}\small\item\em Extract the directory part of a file path. \end{DoxyCompactList}\end{DoxyCompactItemize}
\begin{DoxyCompactItemize}
\item 
typedef struct \hyperlink{structvrna__command__s}{vrna\+\_\+command\+\_\+s} \hyperlink{group__file__utils_ga92cb3b5952352b103bcb32e5a99e0e5a}{vrna\+\_\+cmd\+\_\+t}\hypertarget{group__file__utils_ga92cb3b5952352b103bcb32e5a99e0e5a}{}\label{group__file__utils_ga92cb3b5952352b103bcb32e5a99e0e5a}

\begin{DoxyCompactList}\small\item\em Typename for the command repesenting data structure \hyperlink{structvrna__command__s}{vrna\+\_\+command\+\_\+s}. \end{DoxyCompactList}\end{DoxyCompactItemize}
\begin{DoxyCompactItemize}
\item 
void \hyperlink{group__file__utils_gaaface7db12fadc3d271641c4515ab6e4}{vrna\+\_\+file\+\_\+helixlist} (const char $\ast$seq, const char $\ast$db, float energy, F\+I\+LE $\ast$file)
\begin{DoxyCompactList}\small\item\em Print a secondary structure as helix list. \end{DoxyCompactList}\end{DoxyCompactItemize}


\subsection{Detailed Description}
Functions dealing with file formats for R\+NA sequences, structures, and alignments. 



\subsection{Macro Definition Documentation}
\index{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}!V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+C\+L\+U\+S\+T\+AL@{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+C\+L\+U\+S\+T\+AL}}
\index{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+C\+L\+U\+S\+T\+AL@{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+C\+L\+U\+S\+T\+AL}!Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}}
\subsubsection[{\texorpdfstring{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+C\+L\+U\+S\+T\+AL}{VRNA_FILE_FORMAT_MSA_CLUSTAL}}]{\setlength{\rightskip}{0pt plus 5cm}\#define V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+C\+L\+U\+S\+T\+AL~1U}\hypertarget{group__file__utils_ga79a23de2c7249f2cccd762e475c81859}{}\label{group__file__utils_ga79a23de2c7249f2cccd762e475c81859}


{\ttfamily \#include $<$\hyperlink{file__formats__msa_8h}{Vienna\+R\+N\+A/file\+\_\+formats\+\_\+msa.\+h}$>$}



Option flag indicating ClustalW formatted files. 

\begin{DoxySeeAlso}{See also}
\hyperlink{group__file__utils_gad02d5d12bda54611c915a1019323b7be}{vrna\+\_\+file\+\_\+msa\+\_\+read()}, \hyperlink{group__file__utils_gad4203a438622b2df7bc2f16578d54799}{vrna\+\_\+file\+\_\+msa\+\_\+read\+\_\+record()}, \hyperlink{group__file__utils_gade4fa8136ebb2d0f7eb3f8b59a8658e3}{vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format()} 
\end{DoxySeeAlso}
\index{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}!V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+S\+T\+O\+C\+K\+H\+O\+LM@{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+S\+T\+O\+C\+K\+H\+O\+LM}}
\index{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+S\+T\+O\+C\+K\+H\+O\+LM@{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+S\+T\+O\+C\+K\+H\+O\+LM}!Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}}
\subsubsection[{\texorpdfstring{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+S\+T\+O\+C\+K\+H\+O\+LM}{VRNA_FILE_FORMAT_MSA_STOCKHOLM}}]{\setlength{\rightskip}{0pt plus 5cm}\#define V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+S\+T\+O\+C\+K\+H\+O\+LM~2U}\hypertarget{group__file__utils_ga62be992445cd8ab2ad7a8fded944338b}{}\label{group__file__utils_ga62be992445cd8ab2ad7a8fded944338b}


{\ttfamily \#include $<$\hyperlink{file__formats__msa_8h}{Vienna\+R\+N\+A/file\+\_\+formats\+\_\+msa.\+h}$>$}



Option flag indicating Stockholm 1.\+0 formatted files. 

\begin{DoxySeeAlso}{See also}
\hyperlink{group__file__utils_gad02d5d12bda54611c915a1019323b7be}{vrna\+\_\+file\+\_\+msa\+\_\+read()}, \hyperlink{group__file__utils_gad4203a438622b2df7bc2f16578d54799}{vrna\+\_\+file\+\_\+msa\+\_\+read\+\_\+record()}, \hyperlink{group__file__utils_gade4fa8136ebb2d0f7eb3f8b59a8658e3}{vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format()} 
\end{DoxySeeAlso}
\index{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}!V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+F\+A\+S\+TA@{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+F\+A\+S\+TA}}
\index{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+F\+A\+S\+TA@{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+F\+A\+S\+TA}!Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}}
\subsubsection[{\texorpdfstring{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+F\+A\+S\+TA}{VRNA_FILE_FORMAT_MSA_FASTA}}]{\setlength{\rightskip}{0pt plus 5cm}\#define V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+F\+A\+S\+TA~4U}\hypertarget{group__file__utils_gacf6274a2c825f34e4131404665b00604}{}\label{group__file__utils_gacf6274a2c825f34e4131404665b00604}


{\ttfamily \#include $<$\hyperlink{file__formats__msa_8h}{Vienna\+R\+N\+A/file\+\_\+formats\+\_\+msa.\+h}$>$}



Option flag indicating F\+A\+S\+TA (Pearson) formatted files. 

\begin{DoxySeeAlso}{See also}
\hyperlink{group__file__utils_gad02d5d12bda54611c915a1019323b7be}{vrna\+\_\+file\+\_\+msa\+\_\+read()}, \hyperlink{group__file__utils_gad4203a438622b2df7bc2f16578d54799}{vrna\+\_\+file\+\_\+msa\+\_\+read\+\_\+record()}, \hyperlink{group__file__utils_gade4fa8136ebb2d0f7eb3f8b59a8658e3}{vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format()} 
\end{DoxySeeAlso}
\index{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}!V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+M\+AF@{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+M\+AF}}
\index{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+M\+AF@{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+M\+AF}!Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}}
\subsubsection[{\texorpdfstring{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+M\+AF}{VRNA_FILE_FORMAT_MSA_MAF}}]{\setlength{\rightskip}{0pt plus 5cm}\#define V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+M\+AF~8U}\hypertarget{group__file__utils_gab3186d774c01570d2a47d0308eec4927}{}\label{group__file__utils_gab3186d774c01570d2a47d0308eec4927}


{\ttfamily \#include $<$\hyperlink{file__formats__msa_8h}{Vienna\+R\+N\+A/file\+\_\+formats\+\_\+msa.\+h}$>$}



Option flag indicating M\+AF formatted files. 

\begin{DoxySeeAlso}{See also}
\hyperlink{group__file__utils_gad02d5d12bda54611c915a1019323b7be}{vrna\+\_\+file\+\_\+msa\+\_\+read()}, \hyperlink{group__file__utils_gad4203a438622b2df7bc2f16578d54799}{vrna\+\_\+file\+\_\+msa\+\_\+read\+\_\+record()}, \hyperlink{group__file__utils_gade4fa8136ebb2d0f7eb3f8b59a8658e3}{vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format()} 
\end{DoxySeeAlso}
\index{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}!V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+D\+E\+F\+A\+U\+LT@{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+D\+E\+F\+A\+U\+LT}}
\index{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+D\+E\+F\+A\+U\+LT@{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+D\+E\+F\+A\+U\+LT}!Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}}
\subsubsection[{\texorpdfstring{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+D\+E\+F\+A\+U\+LT}{VRNA_FILE_FORMAT_MSA_DEFAULT}}]{\setlength{\rightskip}{0pt plus 5cm}\#define V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+D\+E\+F\+A\+U\+LT}\hypertarget{group__file__utils_ga4acc255373831856a8417b68de8a94c0}{}\label{group__file__utils_ga4acc255373831856a8417b68de8a94c0}


{\ttfamily \#include $<$\hyperlink{file__formats__msa_8h}{Vienna\+R\+N\+A/file\+\_\+formats\+\_\+msa.\+h}$>$}

{\bfseries Value\+:}
\begin{DoxyCode}
( \(\backslash\)
                                              VRNA\_FILE\_FORMAT\_MSA\_CLUSTAL \(\backslash\)
                                            | \hyperlink{group__file__utils_ga62be992445cd8ab2ad7a8fded944338b}{VRNA\_FILE\_FORMAT\_MSA\_STOCKHOLM} 
      \(\backslash\)
                                            | \hyperlink{group__file__utils_gacf6274a2c825f34e4131404665b00604}{VRNA\_FILE\_FORMAT\_MSA\_FASTA} \(\backslash\)
                                            | \hyperlink{group__file__utils_gab3186d774c01570d2a47d0308eec4927}{VRNA\_FILE\_FORMAT\_MSA\_MAF} \(\backslash\)
                                          )
\end{DoxyCode}


Option flag indicating the set of default file formats. 

\begin{DoxySeeAlso}{See also}
\hyperlink{group__file__utils_gad02d5d12bda54611c915a1019323b7be}{vrna\+\_\+file\+\_\+msa\+\_\+read()}, \hyperlink{group__file__utils_gad4203a438622b2df7bc2f16578d54799}{vrna\+\_\+file\+\_\+msa\+\_\+read\+\_\+record()}, \hyperlink{group__file__utils_gade4fa8136ebb2d0f7eb3f8b59a8658e3}{vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format()} 
\end{DoxySeeAlso}
\index{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}!V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+N\+O\+C\+H\+E\+CK@{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+N\+O\+C\+H\+E\+CK}}
\index{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+N\+O\+C\+H\+E\+CK@{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+N\+O\+C\+H\+E\+CK}!Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}}
\subsubsection[{\texorpdfstring{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+N\+O\+C\+H\+E\+CK}{VRNA_FILE_FORMAT_MSA_NOCHECK}}]{\setlength{\rightskip}{0pt plus 5cm}\#define V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+N\+O\+C\+H\+E\+CK~4096U}\hypertarget{group__file__utils_ga229fb3778ecabea4782902b69fa48fd1}{}\label{group__file__utils_ga229fb3778ecabea4782902b69fa48fd1}


{\ttfamily \#include $<$\hyperlink{file__formats__msa_8h}{Vienna\+R\+N\+A/file\+\_\+formats\+\_\+msa.\+h}$>$}



Option flag to disable validation of the alignment. 

\begin{DoxySeeAlso}{See also}
\hyperlink{group__file__utils_gad02d5d12bda54611c915a1019323b7be}{vrna\+\_\+file\+\_\+msa\+\_\+read()}, \hyperlink{group__file__utils_gad4203a438622b2df7bc2f16578d54799}{vrna\+\_\+file\+\_\+msa\+\_\+read\+\_\+record()} 
\end{DoxySeeAlso}
\index{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}!V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+U\+N\+K\+N\+O\+WN@{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+U\+N\+K\+N\+O\+WN}}
\index{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+U\+N\+K\+N\+O\+WN@{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+U\+N\+K\+N\+O\+WN}!Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}}
\subsubsection[{\texorpdfstring{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+U\+N\+K\+N\+O\+WN}{VRNA_FILE_FORMAT_MSA_UNKNOWN}}]{\setlength{\rightskip}{0pt plus 5cm}\#define V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+U\+N\+K\+N\+O\+WN~8192U}\hypertarget{group__file__utils_gabdc948f547e550125de3e7c65878400c}{}\label{group__file__utils_gabdc948f547e550125de3e7c65878400c}


{\ttfamily \#include $<$\hyperlink{file__formats__msa_8h}{Vienna\+R\+N\+A/file\+\_\+formats\+\_\+msa.\+h}$>$}



Return flag of \hyperlink{group__file__utils_gade4fa8136ebb2d0f7eb3f8b59a8658e3}{vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format()} to indicate unknown or malformatted alignment. 

\begin{DoxySeeAlso}{See also}
\hyperlink{group__file__utils_gade4fa8136ebb2d0f7eb3f8b59a8658e3}{vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format()} 
\end{DoxySeeAlso}


\subsection{Function Documentation}
\index{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}!vrna\+\_\+file\+\_\+helixlist@{vrna\+\_\+file\+\_\+helixlist}}
\index{vrna\+\_\+file\+\_\+helixlist@{vrna\+\_\+file\+\_\+helixlist}!Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}}
\subsubsection[{\texorpdfstring{vrna\+\_\+file\+\_\+helixlist(const char $\ast$seq, const char $\ast$db, float energy, F\+I\+L\+E $\ast$file)}{vrna_file_helixlist(const char *seq, const char *db, float energy, FILE *file)}}]{\setlength{\rightskip}{0pt plus 5cm}void vrna\+\_\+file\+\_\+helixlist (
\begin{DoxyParamCaption}
\item[{const char $\ast$}]{seq, }
\item[{const char $\ast$}]{db, }
\item[{float}]{energy, }
\item[{F\+I\+LE $\ast$}]{file}
\end{DoxyParamCaption}
)}\hypertarget{group__file__utils_gaaface7db12fadc3d271641c4515ab6e4}{}\label{group__file__utils_gaaface7db12fadc3d271641c4515ab6e4}


{\ttfamily \#include $<$\hyperlink{file__formats_8h}{Vienna\+R\+N\+A/file\+\_\+formats.\+h}$>$}



Print a secondary structure as helix list. 


\begin{DoxyParams}{Parameters}
{\em seq} & The R\+NA sequence \\
\hline
{\em db} & The structure in dot-\/bracket format \\
\hline
{\em energy} & Free energy of the structure in kcal/mol \\
\hline
{\em file} & The file handle used to print to (print defaults to \textquotesingle{}stdout\textquotesingle{} if(file == N\+U\+LL) ) \\
\hline
\end{DoxyParams}
\index{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}!vrna\+\_\+file\+\_\+msa\+\_\+read@{vrna\+\_\+file\+\_\+msa\+\_\+read}}
\index{vrna\+\_\+file\+\_\+msa\+\_\+read@{vrna\+\_\+file\+\_\+msa\+\_\+read}!Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}}
\subsubsection[{\texorpdfstring{vrna\+\_\+file\+\_\+msa\+\_\+read(const char $\ast$filename, char $\ast$$\ast$$\ast$names, char $\ast$$\ast$$\ast$aln, char $\ast$$\ast$id, char $\ast$$\ast$structure, unsigned int options)}{vrna_file_msa_read(const char *filename, char ***names, char ***aln, char **id, char **structure, unsigned int options)}}]{\setlength{\rightskip}{0pt plus 5cm}int vrna\+\_\+file\+\_\+msa\+\_\+read (
\begin{DoxyParamCaption}
\item[{const char $\ast$}]{filename, }
\item[{char $\ast$$\ast$$\ast$}]{names, }
\item[{char $\ast$$\ast$$\ast$}]{aln, }
\item[{char $\ast$$\ast$}]{id, }
\item[{char $\ast$$\ast$}]{structure, }
\item[{unsigned int}]{options}
\end{DoxyParamCaption}
)}\hypertarget{group__file__utils_gad02d5d12bda54611c915a1019323b7be}{}\label{group__file__utils_gad02d5d12bda54611c915a1019323b7be}


{\ttfamily \#include $<$\hyperlink{file__formats__msa_8h}{Vienna\+R\+N\+A/file\+\_\+formats\+\_\+msa.\+h}$>$}



Read a multiple sequence alignment from file. 

This function reads the (first) multiple sequence alignment from an input file. The read alignment is split into the sequence id/name part and the actual sequence information and stored in memory as arrays of ids/names and sequences. If the alignment file format allows for additional information, such as an ID of the entire alignment or consensus structure information, this data is retrieved as well and made available. The {\ttfamily options} parameter allows to specify the set of alignment file formats that should be used to retrieve the data. If 0 is passed as option, the list of alignment file formats defaults to \hyperlink{group__file__utils_ga4acc255373831856a8417b68de8a94c0}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+D\+E\+F\+A\+U\+LT}.

Currently, the list of parsable multiple sequence alignment file formats consists of\+:
\begin{DoxyItemize}
\item \hyperlink{file_formats_msa-formats-clustal}{ClustalW format}
\item \hyperlink{file_formats_msa-formats-stockholm}{Stockholm 1.\+0 format}
\item \hyperlink{file_formats_msa-formats-fasta}{F\+A\+S\+TA (Pearson) format}
\item \hyperlink{file_formats_msa-formats-maf}{M\+AF format}
\end{DoxyItemize}\begin{DoxyNote}{Note}
After successfully reading an alignment, this function performs a validation of the data that includes uniqueness of the sequence identifiers, and equal sequence lengths. This check can be deactivated by passing \hyperlink{group__file__utils_ga229fb3778ecabea4782902b69fa48fd1}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+N\+O\+C\+H\+E\+CK} in the {\ttfamily options} parameter.
\end{DoxyNote}
\begin{DoxySeeAlso}{See also}
\hyperlink{group__file__utils_gad4203a438622b2df7bc2f16578d54799}{vrna\+\_\+file\+\_\+msa\+\_\+read\+\_\+record()}, \hyperlink{group__file__utils_ga79a23de2c7249f2cccd762e475c81859}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+C\+L\+U\+S\+T\+AL}, \hyperlink{group__file__utils_ga62be992445cd8ab2ad7a8fded944338b}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+S\+T\+O\+C\+K\+H\+O\+LM}, \hyperlink{group__file__utils_gacf6274a2c825f34e4131404665b00604}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+F\+A\+S\+TA}, \hyperlink{group__file__utils_gab3186d774c01570d2a47d0308eec4927}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+M\+AF}, \hyperlink{group__file__utils_ga4acc255373831856a8417b68de8a94c0}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+D\+E\+F\+A\+U\+LT}, \hyperlink{group__file__utils_ga229fb3778ecabea4782902b69fa48fd1}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+N\+O\+C\+H\+E\+CK}
\end{DoxySeeAlso}

\begin{DoxyParams}{Parameters}
{\em filename} & The name of input file that contains the alignment \\
\hline
{\em names} & An address to the pointer where sequence identifiers should be written to \\
\hline
{\em aln} & An address to the pointer where aligned sequences should be written to \\
\hline
{\em id} & An address to the pointer where the alignment ID should be written to (Maybe N\+U\+LL) \\
\hline
{\em structure} & An address to the pointer where consensus structure information should be written to (Maybe N\+U\+LL) \\
\hline
{\em options} & Options to manipulate the behavior of this function \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
The number of sequences in the alignment, or -\/1 if no alignment record could be found 
\end{DoxyReturn}
\index{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}!vrna\+\_\+file\+\_\+msa\+\_\+read\+\_\+record@{vrna\+\_\+file\+\_\+msa\+\_\+read\+\_\+record}}
\index{vrna\+\_\+file\+\_\+msa\+\_\+read\+\_\+record@{vrna\+\_\+file\+\_\+msa\+\_\+read\+\_\+record}!Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}}
\subsubsection[{\texorpdfstring{vrna\+\_\+file\+\_\+msa\+\_\+read\+\_\+record(\+F\+I\+L\+E $\ast$fp, char $\ast$$\ast$$\ast$names, char $\ast$$\ast$$\ast$aln, char $\ast$$\ast$id, char $\ast$$\ast$structure, unsigned int options)}{vrna_file_msa_read_record(FILE *fp, char ***names, char ***aln, char **id, char **structure, unsigned int options)}}]{\setlength{\rightskip}{0pt plus 5cm}int vrna\+\_\+file\+\_\+msa\+\_\+read\+\_\+record (
\begin{DoxyParamCaption}
\item[{F\+I\+LE $\ast$}]{fp, }
\item[{char $\ast$$\ast$$\ast$}]{names, }
\item[{char $\ast$$\ast$$\ast$}]{aln, }
\item[{char $\ast$$\ast$}]{id, }
\item[{char $\ast$$\ast$}]{structure, }
\item[{unsigned int}]{options}
\end{DoxyParamCaption}
)}\hypertarget{group__file__utils_gad4203a438622b2df7bc2f16578d54799}{}\label{group__file__utils_gad4203a438622b2df7bc2f16578d54799}


{\ttfamily \#include $<$\hyperlink{file__formats__msa_8h}{Vienna\+R\+N\+A/file\+\_\+formats\+\_\+msa.\+h}$>$}



Read a multiple sequence alignment from file handle. 

Similar to \hyperlink{group__file__utils_gad02d5d12bda54611c915a1019323b7be}{vrna\+\_\+file\+\_\+msa\+\_\+read()}, this function reads a multiple sequence alignment from an input file handle. Since using a file handle, this function is not limited to the first alignment record, but allows for looping over all alignments within the input.

The read alignment is split into the sequence id/name part and the actual sequence information and stored in memory as arrays of ids/names and sequences. If the alignment file format allows for additional information, such as an ID of the entire alignment or consensus structure information, this data is retrieved as well and made available. The {\ttfamily options} parameter allows to specify the alignment file format used to retrieve the data. A single format must be specified here, see \hyperlink{group__file__utils_gade4fa8136ebb2d0f7eb3f8b59a8658e3}{vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format()} for helping to determine the correct M\+SA file format.

Currently, the list of parsable multiple sequence alignment file formats consists of\+:
\begin{DoxyItemize}
\item \hyperlink{file_formats_msa-formats-clustal}{ClustalW format}
\item \hyperlink{file_formats_msa-formats-stockholm}{Stockholm 1.\+0 format}
\item \hyperlink{file_formats_msa-formats-fasta}{F\+A\+S\+TA (Pearson) format}
\item \hyperlink{file_formats_msa-formats-maf}{M\+AF format}
\end{DoxyItemize}\begin{DoxyNote}{Note}
After successfully reading an alignment, this function performs a validation of the data that includes uniqueness of the sequence identifiers, and equal sequence lengths. This check can be deactivated by passing \hyperlink{group__file__utils_ga229fb3778ecabea4782902b69fa48fd1}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+N\+O\+C\+H\+E\+CK} in the {\ttfamily options} parameter.
\end{DoxyNote}
\begin{DoxySeeAlso}{See also}
\hyperlink{group__file__utils_gad02d5d12bda54611c915a1019323b7be}{vrna\+\_\+file\+\_\+msa\+\_\+read()}, \hyperlink{group__file__utils_gade4fa8136ebb2d0f7eb3f8b59a8658e3}{vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format()}, \hyperlink{group__file__utils_ga79a23de2c7249f2cccd762e475c81859}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+C\+L\+U\+S\+T\+AL}, \hyperlink{group__file__utils_ga62be992445cd8ab2ad7a8fded944338b}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+S\+T\+O\+C\+K\+H\+O\+LM}, \hyperlink{group__file__utils_gacf6274a2c825f34e4131404665b00604}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+F\+A\+S\+TA}, \hyperlink{group__file__utils_gab3186d774c01570d2a47d0308eec4927}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+M\+AF}, \hyperlink{group__file__utils_ga4acc255373831856a8417b68de8a94c0}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+D\+E\+F\+A\+U\+LT}, \hyperlink{group__file__utils_ga229fb3778ecabea4782902b69fa48fd1}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+N\+O\+C\+H\+E\+CK}
\end{DoxySeeAlso}

\begin{DoxyParams}{Parameters}
{\em fp} & The file pointer the data will be retrieved from \\
\hline
{\em names} & An address to the pointer where sequence identifiers should be written to \\
\hline
{\em aln} & An address to the pointer where aligned sequences should be written to \\
\hline
{\em id} & An address to the pointer where the alignment ID should be written to (Maybe N\+U\+LL) \\
\hline
{\em structure} & An address to the pointer where consensus structure information should be written to (Maybe N\+U\+LL) \\
\hline
{\em options} & Options to manipulate the behavior of this function \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
The number of sequences in the alignment, or -\/1 if no alignment record could be found 
\end{DoxyReturn}
\index{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}!vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format@{vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format}}
\index{vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format@{vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format}!Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}}
\subsubsection[{\texorpdfstring{vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format(const char $\ast$filename, unsigned int options)}{vrna_file_msa_detect_format(const char *filename, unsigned int options)}}]{\setlength{\rightskip}{0pt plus 5cm}unsigned int vrna\+\_\+file\+\_\+msa\+\_\+detect\+\_\+format (
\begin{DoxyParamCaption}
\item[{const char $\ast$}]{filename, }
\item[{unsigned int}]{options}
\end{DoxyParamCaption}
)}\hypertarget{group__file__utils_gade4fa8136ebb2d0f7eb3f8b59a8658e3}{}\label{group__file__utils_gade4fa8136ebb2d0f7eb3f8b59a8658e3}


{\ttfamily \#include $<$\hyperlink{file__formats__msa_8h}{Vienna\+R\+N\+A/file\+\_\+formats\+\_\+msa.\+h}$>$}



Detect the format of a multiple sequence alignment file. 

This function attempts to determine the format of a file that supposedly contains a multiple sequence alignment (M\+SA). This is useful in cases where a M\+SA file contains more than a single record and therefore \hyperlink{group__file__utils_gad02d5d12bda54611c915a1019323b7be}{vrna\+\_\+file\+\_\+msa\+\_\+read()} can not be applied, since it only retrieves the first. Here, one can try to guess the correct file format using this function and then loop over the file, record by record using one of the low-\/level record retrieval functions for the corresponding M\+SA file format.

\begin{DoxyNote}{Note}
This function parses the entire first record within the specified file. As a result, it returns \hyperlink{group__file__utils_gabdc948f547e550125de3e7c65878400c}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+U\+N\+K\+N\+O\+WN} not only if it can\textquotesingle{}t detect the file\textquotesingle{}s format, but also in cases where the file doesn\textquotesingle{}t contain sequences!
\end{DoxyNote}
\begin{DoxySeeAlso}{See also}
\hyperlink{group__file__utils_gad02d5d12bda54611c915a1019323b7be}{vrna\+\_\+file\+\_\+msa\+\_\+read()}, vrna\+\_\+file\+\_\+stockholm\+\_\+read\+\_\+record(), vrna\+\_\+file\+\_\+clustal\+\_\+read\+\_\+record(), \hyperlink{file__formats_8h_a8cfb7e271efc9e1f34640acb85475639}{vrna\+\_\+file\+\_\+fasta\+\_\+read\+\_\+record()}
\end{DoxySeeAlso}

\begin{DoxyParams}{Parameters}
{\em filename} & The name of input file that contains the alignment \\
\hline
{\em options} & Options to manipulate the behavior of this function \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
The M\+SA file format, or \hyperlink{group__file__utils_gabdc948f547e550125de3e7c65878400c}{V\+R\+N\+A\+\_\+\+F\+I\+L\+E\+\_\+\+F\+O\+R\+M\+A\+T\+\_\+\+M\+S\+A\+\_\+\+U\+N\+K\+N\+O\+WN} 
\end{DoxyReturn}
\index{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}!vrna\+\_\+read\+\_\+line@{vrna\+\_\+read\+\_\+line}}
\index{vrna\+\_\+read\+\_\+line@{vrna\+\_\+read\+\_\+line}!Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments@{Functions to Read/\+Write several File Formats for R\+N\+A Sequences, Structures, and Alignments}}
\subsubsection[{\texorpdfstring{vrna\+\_\+read\+\_\+line(\+F\+I\+L\+E $\ast$fp)}{vrna_read_line(FILE *fp)}}]{\setlength{\rightskip}{0pt plus 5cm}char$\ast$ vrna\+\_\+read\+\_\+line (
\begin{DoxyParamCaption}
\item[{F\+I\+LE $\ast$}]{fp}
\end{DoxyParamCaption}
)}\hypertarget{group__file__utils_gad16c270b70a77c753088b29de8a802dc}{}\label{group__file__utils_gad16c270b70a77c753088b29de8a802dc}


{\ttfamily \#include $<$\hyperlink{file__utils_8h}{Vienna\+R\+N\+A/file\+\_\+utils.\+h}$>$}



Read a line of arbitrary length from a stream. 

Returns a pointer to the resulting string. The necessary memory is allocated and should be released using {\itshape free()} when the string is no longer needed.


\begin{DoxyParams}{Parameters}
{\em fp} & A file pointer to the stream where the function should read from \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
A pointer to the resulting string 
\end{DoxyReturn}
